src/common/buffer.cc

```diff
@@ -2132,26 +2120,10 @@ buffer::ptr_node::create_hypercombined(ceph::unique_leakable_ptr<buffer::raw> r)
     new ptr_node(std::move(r)));
 }
 
-buffer::ptr_node* buffer::ptr_node::copy_hypercombined(
-  const buffer::ptr_node& copy_this)
-{
-  // FIXME: we don't currently hypercombine buffers due to crashes
-  // observed in the rados suite. After fixing we'll use placement
-  // new to create ptr_node on buffer::raw::bptr_storage.
-  auto raw_new = copy_this._raw->clone();
-  return new ptr_node(copy_this, std::move(raw_new));
-}
-
 buffer::ptr_node* buffer::ptr_node::cloner::operator()(
   const buffer::ptr_node& clone_this)
 {
-  const raw* const raw_this = clone_this._raw;
-  if (likely(!raw_this || raw_this->is_shareable())) {
-    return new ptr_node(clone_this);
-  } else {
-    // clone non-shareable buffers (make shareable)
-   return copy_hypercombined(clone_this);
-  }
+  return new ptr_node(clone_this);
```

Comment:
@rzarzynski any plan to fix the hypercombing? if not, i'd suggest to drop it at this moment.

Reply:
I count on the current memory corruption issue was responsible also the unidentified failures which caused disabling HC in 2019.